ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/ClusterTerminationTest.java

```diff
@@ -0,0 +1,207 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.integration;
+
+import static io.confluent.ksql.serde.DataSource.DataSourceSerDe.JSON;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.common.utils.IntegrationTest;
+import io.confluent.ksql.integration.IntegrationTestHarness;
+import io.confluent.ksql.rest.client.KsqlRestClient;
+import io.confluent.ksql.rest.client.RestResponse;
+import io.confluent.ksql.rest.entity.ClusterTerminateRequest;
+import io.confluent.ksql.rest.entity.CommandStatus;
+import io.confluent.ksql.rest.entity.CommandStatus.Status;
+import io.confluent.ksql.rest.entity.CommandStatusEntity;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.rest.entity.KsqlEntityList;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.rest.server.TestKsqlRestApp;
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.PageViewDataProvider;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+
+@SuppressWarnings("unchecked")
+@Category({IntegrationTest.class})
+public class ClusterTerminationTest {
+
+  private static final String PAGE_VIEW_TOPIC = "pageviews";
+  private static final String PAGE_VIEW_STREAM = "pageviews_original";
+  private static final PageViewDataProvider PAGE_VIEW_DATA_PROVIDER = new PageViewDataProvider();
+
+  private static final String SINK_TOPIC = "sink_topic";
+  private static final String SINK_STREAM = "sink_stream";
+
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .buildWithServiceContext(TEST_HARNESS::serviceContext);
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain.outerRule(TEST_HARNESS).around(REST_APP);
+
+  private KsqlRestClient restClient;
+
+  @BeforeClass
+  public static void setUpClass() {
+    TEST_HARNESS.ensureTopics(PAGE_VIEW_TOPIC);
+
+    createStreams(PAGE_VIEW_STREAM);
+  }
+
+  @Before
+  public void setUp() {
+    restClient = REST_APP.buildKsqlClient();
+  }
+
+  @After
+  public void cleanUp() {
+    restClient.close();
+  }
+
+  @Test
+  public void shouldCleanUpSinkTopicsAndSchemasDuringClusterTermination() throws Exception {
+    // Given:
+    makeKsqlRequest("CREATE STREAM " + SINK_STREAM
```

Comment:
I'm not convinced of the value of having a test for the case where multiple sources are registered with a single topic, since people shouldn't be declaring multiple CSAS/CTAS statements with sink topics backed by the same topic. (If they wanted to achieve that, they should use CSAS/CTAS for the first statement and INSERT INTO for subsequent statements.) It seems plausible someone would want to declare multiple sources on top of the same topic via CREATE STREAM/TABLE statements, but you can't drop those from cluster termination, only sinks.

Also, there are a number of additional integration tests we should add around the cluster termination functionality, e.g., tests to verify that the command topic and processing log topics are cleaned up, and tests to verify that internal topics are cleaned up too. I'm holding off on adding those tests into this PR in order to keep the PR more targeted, but if people think there's value in getting those tests into 5.2, I can add them soon.

Reply:
> It seems plausible someone would want to declare multiple sources on top of the same topic via CREATE STREAM/TABLE statements, but you can't drop those from cluster termination, only sinks.

Ah yes, this makes a lot of sense (I was only referring to CT/CS)